### ISSUE TITLE
FR: Ability to use RecapchaVerifier in other use cases besides phone auth

### ISSUE THREAD

**Issue by sluramod** [firebase/firebase-js-sdk#4042]
### [REQUIRED] Describe your environment

  * Operating System version: any
  * Browser version: any
  * Firebase SDK version: 7.x
  * Firebase Product: auth

#### Context

Sometimes it's necessary to protect cloud functions by recaptcha.

Firebase has RecaptchaVerifier implementation that is used in Sign in with phone number. Documentation states that site key and secret key are automatically generated for the project. RecaptchaVerifier accepts all parameters EXCEPT siteKey, because it's automatically generated. There is explicit check that throws error if siteKey parameter is specified.

On the server, in order to verify Recaptcha response it's necessary to have secret key that matches siteKey that was used on the client.

#### Question

Is there a way to get automatically generated secret key that matches site key used by RecaptchaVerifier on the server in a cloud function?

#### Proposed Solution

Allow clients to specify siteKey when creating instance of RecaptchaVerifier so that it can be used to protect any server function.

**Comment by Feiyang1** [comment]
I don't think the auto-generated recaptcha key is meant to be shared and reused outside of auth. To protect other parts of your app, you can use your own recaptcha account. I believe recaptcha supports multiple instances in the same site.

@avolkovi Does it sound accurate?

**Comment by avolkovi** [comment]
I believe recaptcha tokens are single use, but you can verify them server side https://developers.google.com/recaptcha/docs/verify to protect your Cloud Function

**Comment by sluramod** [comment]
> I believe recaptcha tokens are single use, but you can verify them server side https://developers.google.com/recaptcha/docs/verify to protect your Cloud Function

@avolkovi I would love to verify them, but I don’t know what secret key to use. 

@Feiyang1 I would love to specify my own site key when creating instance of RecaptchaVerifier, but that parameter is explicitly not allowed. The only way to use my own recaptcha site key is to duplicate all code of RecaptchaVerifier. The new code will be 99% same except it will allow to specify my own site key.

**Comment by avolkovi** [comment]
@sluramod you're right, since we manage the recapatcha config for you you can't verify on your own... so one option would be to let you specify a custom recaptcha config for your project instead of using the one we manage.

I'll bring your use case up with our product team, can't promise timelines unfortunately.

**Comment by sluramod** [comment]
@avolkovi @Feiyang1 any progress on this?

**Comment by sluramod** [comment]
@avolkovi @MeghaB @Feiyang1 any updates on this?

**Comment by bentacos** [comment]
please allow us to use our own recaptcha sitekey. there are cases that we want to verify both recaptcha response and auth token.

**Comment by neekey** [comment]
any progress on this?
